### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by ZacSweers**
This includes consolidating on Okio 2.x now that we're moving to Kotlin here. Note that I've held off on KotlinPoet due to https://github.com/square/kotlinpoet/issues/999

### CODE REVIEW

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 1]
```diff
@@ -19,8 +19,7 @@ object Dependencies {
   const val asm = "org.ow2.asm:asm:7.1"
   const val jsr305 = "com.google.code.findbugs:jsr305:3.0.2"
   const val ktlintVersion = "0.39.0"
-  const val okio = "com.squareup.okio:okio:1.17.5"
-  const val okio2 = "com.squareup.okio:okio:2.8.0"
+  const val okio = "com.squareup.okio:okio:2.9.0"
```
Comment (ZacSweers) [inline buildSrc/src/main/kotlin/Dependencies.kt:22]:
```suggestion
  const val okio = "com.squareup.okio:okio:2.10.0"
```

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 2]
```diff
@@ -50,8 +49,8 @@ object Dependencies {
 
   object Testing {
     const val assertj = "org.assertj:assertj-core:3.11.1"
-    const val compileTesting = "com.github.tschuchortdev:kotlin-compile-testing:1.2.11"
-    const val junit = "junit:junit:4.13"
+    const val compileTesting = "com.github.tschuchortdev:kotlin-compile-testing:1.3.1"
```
Comment (ZacSweers) [inline buildSrc/src/main/kotlin/Dependencies.kt:52]:
```suggestion
    const val compileTesting = "com.github.tschuchortdev:kotlin-compile-testing:1.3.4"
```

**Review by ShaishavGandhi — APPROVED** [review]

**Review by Egorand — APPROVED** [review]
KotlinPoet issue has been solved, wanna bump to 1.7.x?

**Comment by ZacSweers** [comment]
Handled in a separate PR!

**Pull request merged**